pkg/apis/core/v1/helper/helpers.go

```diff
@@ -413,6 +413,28 @@ func TolerationsTolerateTaintsWithFilter(tolerations []v1.Toleration, taints []v
 	return true
 }
 
+// TaintsNotToleratedWithFilter checks if the given tolerations tolerates
+// all the filtered taints, and returns the not tolerated taints
+func TaintsNotToleratedWithFilter(taints []v1.Taint, tolerations []v1.Toleration, applyFilter taintsFilterFunc) []v1.Taint {
```

Comment:
I think we want something simpler than that, by design the scheduler stops after the first failed Filter, and we should keeping doing the same here: stop after the first non matching taint, and just return that taint.

Reply:
changed this.